MetadataExtractor/Formats/Exif/makernotes/ReconyxHyperFire2MakernoteDescriptor.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace MetadataExtractor.Formats.Exif.Makernotes
+{
+    /// <summary>
+    /// Provides human-readable string representations of tag values stored in a <see cref="ReconyxHyperFire2MakernoteDirectory"/>.
+    /// </summary>
+    /// <remarks>Reconyx uses a fixed makernote block. Tag values are the byte index of the tag within the makernote.</remarks>
+    [SuppressMessage("ReSharper", "MemberCanBePrivate.Global")]
+    public sealed class ReconyxHyperFire2MakernoteDescriptor : TagDescriptor<ReconyxHyperFire2MakernoteDirectory>
+    {
+        public ReconyxHyperFire2MakernoteDescriptor(ReconyxHyperFire2MakernoteDirectory directory)
+            : base(directory)
+        {
+        }
+
+        public override string? GetDescription(int tagType)
+        {
+            switch (tagType)
+            {
+                case ReconyxHyperFire2MakernoteDirectory.TagFileNumber:
+                    return Directory.GetUInt16(tagType).ToString();
+                case ReconyxHyperFire2MakernoteDirectory.TagDirectoryNumber:
+                    return Directory.GetUInt16(tagType).ToString();
+                case ReconyxHyperFire2MakernoteDirectory.TagFirmwareVersion:
+                    var version = Directory.GetObject(tagType) as Version;
+                    if (version != null)
+                        return $"{version.Major}.{version.Minor}{(char)version.Build}";
+                    else
+                        return string.Empty;
+                case ReconyxHyperFire2MakernoteDirectory.TagFirmwareDate:
+                    return Directory.GetDateTime(tagType).ToString(CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern);
+                case ReconyxHyperFire2MakernoteDirectory.TagTriggerMode:
+                    return Directory.GetString(tagType) switch
+                    {
+                        "M" => "Motion Detection",
+                        "P" => "Point and Shoot",
+                        "T" => "Time Lapse",
+                        _ => "Unknown trigger mode",
+                    };
+                case ReconyxHyperFire2MakernoteDirectory.TagSequence:
+                    var sequence = Directory.GetInt32Array(tagType);
+                    return sequence != null ? $"{sequence[0]}/{sequence[1]}" : base.GetDescription(tagType);
```

Comment:
```suggestion
                    return sequence != null && sequence.Length > 1 ? $"{sequence[0]}/{sequence[1]}" : base.GetDescription(tagType);
```

Reply:
The makernote reader will always have a two element array in here, but I want to think about a future where we let people set their own tag values.